### PR TITLE
Added Instagram Provider

### PR DESCRIPTION
**Pull request by gbaldera**
Added Instagram Provider following the docs from http://instagram.com/developer/

### CODE REVIEW

File: libraries/providers/instagram.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+		// Create a response from the request
+		return array(
+			'uid' => $user->id,
+			'nickname' => $user->username,
+			'name' => $user->full_name,
+			'email' => null,
+			'profile_picture' => $user->profile_picture,
+			'bio' => $user->bio,
+            'counts' => array(
```
Comment (philsturgeon) [inline libraries/providers/instagram.php:43]: Can you fix the alignment here. Also, we don't need all these fields.

Can you try to match the field return up to other fields? Return separate names if you can, ignore follows and things, and make sure `profile_picture` is the right key.

Comment (gbaldera) [inline libraries/providers/instagram.php:43]: Sure no problem

**Comment by philsturgeon** [comment]
In fact I've already done an Instagram provider for the FuelPHP OAuth2 cell. It is ridiculous that they are two separate codebases as they're almost identical, but one is PHP 5.2 and one is PHP 5.3 :-/

**Comment by gbaldera** [comment]
It's ok I just read the readme about Instagram but didn't see the provider in the providers directory so I made this. :)

**Pull request closed without merging**